### PR TITLE
A wider website and documentation navigation revamp

### PR DESCRIPTION
**Pull request by Madelena**
## Proposed change
This enables an extra wide website responsive design that was previously hidden from the theme. With the extra space, we can show navigation on the page much more consistently.

I also used the opportunity to consolidate and clean up the navigation of the documentation sections.

image
image
image
image

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
  - Enabled the table of contents (TOC) for dashboards, integrations, and various documentation sections.

- **Style**
  - Improved typography, alignment, and responsiveness across different devices.
  - Updated styling for grid layouts, menus, and documentation index.

- **Documentation**
  - Restructured the documentation sitemap with new sections and icons for better navigation.
  - Added a recent blog post categories section to the website template.

- **User Experience**
  - Enhanced navigation in the "Getting Started" section with more specific sub-menu items.

### CODE REVIEW

File: source/_includes/asides/docs_sitemap.html [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+</li>
+
+<li>
+  {% icon "mdi:gauge" %} {% active_link /dashboards Dashboards %}
+  {% if root == 'dashboards' or include.docs_index %}
+  <ul>
+    {% assign elements = site.dashboards | sort_natural: 'title' %}
+
+    <li>
+      <iconify-icon icon="mdi:about-circle"></iconify-icon> Dashboard Basics
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html]:
```suggestion
      <iconify-icon icon="mdi:about-circle"></iconify-icon> Dashboard basics
```

File: source/_includes/asides/docs_sitemap.html [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+          </a></li>
+      </ul>
+    </li>
+
+  </ul>
+  {% endif %}
+</li>
+
+<li>
+  {% icon "mdi:microphone" %} {% active_link /voice_control/ Voice Assistants %}
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html]:
```suggestion
  {% icon "mdi:microphone" %} {% active_link /voice_control/ Voice assistants %}
```

File: source/_includes/asides/docs_sitemap.html [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    <li>{% active_link /docs/organizing/floors/ Floors %}</li>
+    <li>{% active_link /docs/organizing/labels/ Labels %}</li>
+    <li>{% active_link /docs/organizing/categories/ Categories %}</li>
+    <li>{% active_link /docs/frontend/icons/ Icons %}</li>
+  </ul>
+  {% endif %}
+</li>
+
+<li>
+  {% icon "mdi:solar-power-variant" %} {% active_link /docs/energy/ Home Energy Management %}
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html]:
```suggestion
  {% icon "mdi:solar-power-variant" %} {% active_link /docs/energy/ Home energy management %}
```

File: source/_includes/asides/docs_sitemap.html [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+    <li>{% active_link /docs/energy/electricity-grid/ Electricity grid %}</li>
+    <li>{% active_link /docs/energy/solar-panels/ Solar panels %}</li>
+    <li>{% active_link /docs/energy/individual-devices/ Individual devices %}</li>
+    <li>{% active_link /docs/energy/faq/ FAQ %}</li>
+  </ul>
+  {% endif %}
+</li>
+
+<li>
+  {% icon "mdi:cogs" %} {% active_link /docs/configuration/ Advanced Configuration %}
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html]:
```suggestion
  {% icon "mdi:cogs" %} {% active_link /docs/configuration/ Advanced configuration %}
```

File: source/_includes/asides/docs_sitemap.html [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+    <li>{% active_link /docs/configuration/events/ Events %}</li>
+    <li>{% active_link /docs/configuration/state_object/ State objects %}</li>
+    <li>{% active_link /docs/configuration/templating/ Templating %}</li>
+    <li>{% active_link /docs/configuration/platform_options/ Entity component platform options %}</li>
+  </ul>
+  {% endif %}
+</li>
+
+<li>
+  {% icon "mdi:update" %} {% active_link /common-tasks/os/ Common Tasks %}
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html]:
```suggestion
  {% icon "mdi:update" %} {% active_link /common-tasks/os/ Common tasks %}
```

File: source/_includes/asides/docs_sitemap.html [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+  </ul>
+  {% endif %}
+</li>
+
+<li>
+  {% icon "mdi:account-security" %} {% active_link /docs/authentication/ Authentication %}
+  {% if doc == 'authentication' or include.docs_index %}
+  <ul>
+    <li>{% active_link /docs/authentication/providers/ Auth providers %}</li>
+    <li>{% active_link /docs/authentication/multi-factor-auth/ Multi factor auth %}</li>
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html]:
```suggestion
    <li>{% active_link /docs/authentication/multi-factor-auth/ Multi-factor authentication %}</li>
```

File: source/_includes/asides/docs_sitemap.html [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+    <li>{% active_link /docs/locked_out/ I'm locked out %}</li>
+  </ul>
+  {% endif %}
+</li>
+
+<li>
+  {% icon "mdi:database" %} {% active_link /docs/backend/ Backend %}
+  {% if doc == 'backend' or include.docs_index %}
+  <ul>
+    <li>{% active_link /docs/frontend/ User Interface %}</li>
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html]:
```suggestion
    <li>{% active_link /docs/frontend/ User interface %}</li>
```

File: source/_includes/asides/docs_sitemap.html [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+    <li>{% active_link /docs/tools/dev-tools/ Developer tools %}</li>
+    <li>{% active_link /docs/tools/quick-bar/ Quick bar %}</li>
+    <li>{% active_link /docs/tools/hass/ hass %}</li>
+    <li>{% active_link /docs/tools/check_config/ check_config %}</li>
+  </ul>
+  {% endif %}
+</li>
+
+<li>
+  {% icon "mdi:cellphone" %} <a href="https://companion.home-assistant.io/docs">iOS and Android Apps {% icon "tabler:external-link" %}</a>
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html]:
```suggestion
  {% icon "mdi:cellphone" %} <a href="https://companion.home-assistant.io/docs">iOS and Android apps {% icon "tabler:external-link" %}</a>
```

File: source/_includes/asides/docs_sitemap.html [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+    <li>{% active_link /integrations/mqtt/#publish--dump-services Publish service %}</li>
+    <li>{% active_link /integrations/mqtt/#birth-and-last-will-messages Birth and last will messages %}</li>
+    <li>{% active_link /integrations/mqtt/#testing-your-setup Testing your setup %}</li>
+    <li>{% active_link /integrations/mqtt/#logging Logging %}</li>
+  </ul>
+  {% endif %}
+</li>
+
+<li>
+  {% icon "bitcoin-icons:node-hardware-filled" %} Official Hardware
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html]:
```suggestion
  {% icon "bitcoin-icons:node-hardware-filled" %} Official hardware
```

File: source/_includes/asides/recent_categories.html [hunk 10]
```diff
@@ -0,0 +1,14 @@
+{% capture recent_categories %}{% for post in site.posts limit: 20 %}{{ post.categories | join: ", " }}, {% endfor %}{% endcapture %}
+{% assign recent_categories = recent_categories | split: ", " | uniq | compact %}
+
+<section class="recent-categories aside-module grid__item one-whole lap-one-half">
+  <h1 class="title epsilon">{% icon "mdi:category" %}  Recent Categories</h1>
```
Comment (c0ffeeca7) [inline source/_includes/asides/recent_categories.html]:
```suggestion
  <h1 class="title epsilon">{% icon "mdi:category" %}  Recent categories</h1>
```

File: source/_includes/asides/docs_sitemap.html [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+    <li>{% active_link /docs/configuration/secrets/ Storing secrets %}</li>
+    <li>{% active_link /docs/configuration/events/ Events %}</li>
+    <li>{% active_link /docs/configuration/state_object/ State objects %}</li>
+    <li>{% active_link /docs/configuration/templating/ Templating %}</li>
+    <li>{% active_link /docs/configuration/platform_options/ Entity component platform options %}</li>
+  </ul>
+  {% endif %}
+</li>
+
+<li>
```
Comment (c0ffeeca7) [inline source/_includes/asides/docs_sitemap.html:218]: should common tasks maybe be placed further up, above Advanced configuration? It covers tasks such as update and backup, as well as configuring file access and setting up an editor, which is used for advanced config.

**Comment by c0ffeeca7** [comment]
Nice :star_struck:

File: source/_includes/asides/docs_sitemap.html (hunk 11 above)
Comment (Madelena) [reply source/_includes/asides/docs_sitemap.html:218]: Sure, that makes sense. 😊 This PR is mostly about the UI so I tried to make as few editorial decisions as I could.

Comment (frenck) [reply source/_includes/asides/docs_sitemap.html:218]: Done in [baca135](https://github.com/home-assistant/home-assistant.io/pull/33601/commits/baca135471f19d298f613e1bc41b8a696c929e90)

**Review by frenck — APPROVED** [review]
Some small cleanups done, otherwise: Nice! 🤩

**Review by c0ffeeca7 — COMMENTED** [review]
Thank you, @Madelena :+1:

**Comment by Madelena** [comment]
Wooop Woop! 🤩

**Pull request merged**